### DISCUSSION TITLE
Is there a way to format numbers in the status bar as zero-padded?

### DISCUSSION THREAD

**Discussion by rbanffy** [tmux/tmux#3849] [General]
If I format a number (3 in this case) with p (as in `#{p-3:@v]`, it'll be output as `  3`. Is there an option to render it as `003`?

**Comment by nicm** [comment]
I think you'd need to do it with comparisons maybe (not tested) `#{?#{e|<|:#{@v},10},0#{@v},#{@v}}`.

Would not be difficult to add a flag to `p` to use zeros.

**Reply by rbanffy** [reply]
What would a zero-padded "p" option look like? I'm not very familiar with the micro-language in there, so I have very little idea of what it should look in order to be easy to use.

**Reply by nicm** [reply]
A flag would be best as the second argument like the `r` flag to `m` so you would do for example `#{p|3|z:@v}`. I don't think it would be difficult to add but I can't check now - look at `format.c`.
